### DISCUSSION TITLE
Is there any way to use streamlit on the Planetary Computer Hub?

### DISCUSSION THREAD

**Discussion by wiamadnan** [microsoft/PlanetaryComputer#325] [Q&A]
How can I can open streamlit app 

```
PlanetaryComputerExamples/hackthon_model/src$ streamlit run app.py

Collecting usage statistics. To deactivate, set browser.gatherUsageStats to False.


  You can now view your Streamlit app in your browser.

  Network URL: http://10.244.78.8:8501
  External URL: http://20.31.33.22:8501
```

**Comment by TomAugspurger** [comment]
I've never used Streamlit before, but for similar types of applications you can use jupyter-server-proxy<https://github.com/jupyterhub/jupyter-server-proxy> to proxy the port to on the notebook server to your browser.
________________________________
From: WiamAdnan ***@***.***>
Sent: Wednesday, February 28, 2024 6:07 AM
To: microsoft/PlanetaryComputer ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [microsoft/PlanetaryComputer] Is there any way to use streamlit on the Planetary Computer Hub? (Discussion #325)


How can I can open streamlit app

PlanetaryComputerExamples/hackthon_model/src$ streamlit run app.py

Collecting usage statistics. To deactivate, set browser.gatherUsageStats to False.


  You can now view your Streamlit app in your browser.

  Network URL: http://10.244.78.8:8501
  External URL: http://20.31.33.22:8501
